src/services/drpcEvents/index.ts

```diff
@@ -146,4 +153,85 @@ export default class DrpcEvents {
       })
     }
   }
+
+  private async handleSubmitQueryResponse(request: DrpcRequest, agentConnectionId: string) {
+    let queryId: string | null = null
+    try {
+      this.logger.info(
+        'DRPC response (%s) received on connection %s of method %s',
+        request.id,
+        agentConnectionId,
+        request.method
+      )
+
+      let params: SubmitQueryRPCParams
+      try {
+        params = submitQueryRpcParams.parse(request.params)
+      } catch (err) {
+        this.logger.warn('Invalid parameters received for request %s: %o', request.id, request.params)
+        await this.cloudagent.submitDrpcResponse(request.id, {
+          error: {
+            code: drpcErrorCode.INVALID_PARAMS,
+            message: `invalid params object`,
+          },
+        })
+        return
+      }
+      const [connection] = await this.db.get('connection', { agent_connection_id: agentConnectionId })
+      if (!connection) {
+        this.logger.warn('Invalid connection for drpc message %s', agentConnectionId)
+        return
+      }
+      //find corresponding query based on queryIdForResponse provided by responder,
+      //this should match a query id in our db
+      const [queryRow] = await this.db.get('query', { id: params.queryIdForResponse })
+      if (!connection) {
+        this.logger.warn('Invalid queryId for drpc message %s', params.queryIdForResponse)
+        return
+      }
+      //update corresponding query
+      this.logger.warn(queryRow)
+      const [query] = await this.db.update(
+        'query',
+        { id: queryRow.id },
+        { query_response: params.emissions, status: 'resolved' }
```

Comment:
I think we can simplify this to `{ response: params.emissions, status: 'resolved' }`

Reply:
but we need to make sure that the status has been updated? Unless we want to remove the status field?

Reply:
yep misread. Leave as is